File: src/Group.java
import java.util.Vector;

public class Group extends Object3D {
	Vector<Object3D> vector = null;
	
	public Group() {
		vector = new Vector<Object3D>();
	}
	
	public void add(Object3D o3D) {
		vector.add(o3D);
	}
	
	boolean intersect(Ray r, Hit h, Range range, TValue tMin) {
		// ToDo:
		// Iterate over the array of objects,
		// finding the intersections
		boolean retVal = false;
		// ...
		return retVal;
	}

}

File: src/Group.scala
import java.util.Vector;
import javax.vecmath.Color3f;

case class Group (var vector:List [SceneElement] )
{

      
    
      def intersect( ray:Ray ):Hit =
        {
          var hit     = new Hit (-1,new Color3f ());
          var range   = new Range ();

          vector.foreach {            
            case  SceneObject (g,Material (pigment)) =>

              g match {
                case s:Sphere  => {

                  s.intersect(ray,hit,range,pigment) match{
                    case Some ((nh,nr)) =>
                      {
                        hit   = nh;
                        range = nr;
                        
                      }
                    case _ => ()
                  }
                }
                case _ => ()
              }
            case _ => ()
          }
          
            // ToDo:
           // Iterate over the array of objects,
         	// finding the intersections
	
         	// ...
         	return hit;
         }
   	

}
